app/src/main/java/com/github/multimatum_team/multimatum/AddDeadlineActivity.kt

```diff
@@ -1,57 +1,103 @@
 package com.github.multimatum_team.multimatum
 
-import androidx.appcompat.app.AppCompatActivity
+import android.app.DatePickerDialog
+import android.app.DatePickerDialog.OnDateSetListener
+import android.app.TimePickerDialog
 import android.os.Bundle
 import android.view.View
-import android.widget.DatePicker
 import android.widget.TextView
 import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
 import com.github.multimatum_team.multimatum.model.Deadline
 import com.github.multimatum_team.multimatum.model.DeadlineState
-import com.github.multimatum_team.multimatum.repository.DeadlineRepository
+import com.github.multimatum_team.multimatum.viewmodel.DeadlineListViewModel
 import dagger.hilt.android.AndroidEntryPoint
-import kotlinx.coroutines.runBlocking
-import java.time.LocalDate
+import java.time.Instant
 import java.time.LocalDateTime
-import javax.inject.Inject
+import java.time.ZoneId
+import java.time.temporal.ChronoUnit
+
 
 /**
- *  Activity who create a deadline using a DatePicker
+ *  Activity who create a deadline using a DatePickerDialog and a TimePickerDialog
  */
 @AndroidEntryPoint
 class AddDeadlineActivity : AppCompatActivity() {
 
-    companion object {
-        // Temporary value that will be link later to the actual day
-        private val initDate : LocalDateTime = LocalDateTime.of(2022,1,1, 0, 0)
-    }
+    private var selectedDate: LocalDateTime =
+        Instant.now().atZone(ZoneId.systemDefault()).toLocalDateTime().truncatedTo(ChronoUnit.HOURS)
 
-    @Inject
-    lateinit var deadlineRepository: DeadlineRepository
+    private val viewModel: DeadlineListViewModel by viewModels()
+    private lateinit var textDate: TextView
+    private lateinit var textTime: TextView
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_add_deadline)
-        val datePicker = findViewById<DatePicker>(R.id.add_deadline_date_picker)
-        datePicker.init(initDate.year,initDate.month.value,initDate.dayOfMonth, null)
+        textDate = findViewById(R.id.add_deadline_text_date)
+        textTime = findViewById(R.id.add_deadline_text_time)
+        textDate.text = selectedDate.toLocalDate().toString()
+        textTime.text = selectedDate.toLocalTime().toString()
+    }
+
+    //Setup a DatePickerDialog that will select a date for the deadline and show it
```

Comment:
Here you should use proper comment format for functions (/** ... */) or make it a private function.